File: Python/dsaOne/online-assessments-practice/getMinSum.js
const getMinSum = (sec_val, msg) => {
    let list = []

    for(let i =0; i < msg.length; i++){
        let idx = parseInt(msg[i].charCodeAt(0) - 'a'.charCodeAt(0))
        let val = sec_val[idx]
        list.push(val)
    }
    
    list.sort((a,b) => a - b)
    console.log('list seen', list)
    
    let res = []
    for(let j = 0; j < list.length-1; j++) {
        res = res + list[j+1] - list[j]
    }
    return res
}

console.log(getMinSum([1,2,1,3,1,3,5,7,1,1], 'afeb'))

File: Python/dsaOne/online-assessments-practice/getMinSum.py
def getMinSum(sec_val, msg):
    li = []

    for x in msg:
        ind = ord(x) - ord("a")
        val = sec_val[ind]
        li.append(val)

    print('li seen', li)
    li.sort()
    

    res = 0
    for i in range(len(li)-1):
        res = res + (li[i+1] - li[i])

    return res

print(getMinSum([1,2,1,3,1,3,5,7,1,1], 'afeb'))
